tools/cfn2ts/lib/codegen.ts

```diff
@@ -450,27 +467,31 @@ export default class CodeGenerator {
 
       const self = this;
       const validatorExpression = genspec.typeDispatch<string>(resource, propSpec, {
-        visitScalar(type: genspec.CodeName) {
-          return  validatorNames([type]);
+        visitScalar(type: genspec.CodeName, recurse) {
+          const specType = type.specName && self.spec.PropertyTypes[type.specName.fqn];
+          if (specType && !schema.isPropertyBag(specType)) {
+            return recurse(specType);
```

Comment:
Since this is the only place where we actually recurse, and the function we want to recurse into is in scope (it's `genspec.typeDispatch<string>(resource, propsSpec, objectWithCallbacks)`... why not just make the recursive call here and avoid adding the `recurse` argument to all visitors?

Reply:
Yeah that's fair. Somehow when I started this I felt like it could apply in more places... But there definitely isn't any reason why.